docs/Consensus/ibft/ibft.md

```diff
@@ -0,0 +1,184 @@
+    - Validator receives `ceil(2N/3)` of valid `PREPARE` messages to enter `PREPARED` state. Valid messages conform to the following conditions:
+        - Matched sequence and round.
+        - Matched block hash.
+        - Messages are from known validators.
+    - Validator broadcasts `COMMIT` message upon entering `PREPARED` state.
+- `PREPARED` -> `COMMITTED`:
+    - **Validator** receives `ceil(2N/3)` of valid `COMMIT` messages to enter `COMMITTED` state. Valid messages conform to the following conditions:
+        - Matched sequence and round.
+        - Matched block hash.
+        - Messages are from known validators.
+- `COMMITTED` -> `FINAL COMMITTED`:
+    - **Validator** appends `ceil(2N/3)` commitment signatures to `extraData` and tries to insert the block into the blockchain.
+    - **Validator** enters `FINAL COMMITTED` state when insertion succeeds.
+- `FINAL COMMITTED` -> `NEW ROUND`:
+    - **Validators** pick a new **proposer** and begin a new round timer.
+
+#### Round change flow
+
+- There are three conditions that would trigger `ROUND CHANGE`:
+    - Round change timer expires.
+    - Invalid `PREPREPARE` message.
+    - Block insertion fails.
+- When a validator notices that one of the above conditions applies, it broadcasts a `ROUND CHANGE` message along with the proposed round number and waits for `ROUND CHANGE` messages from other validators. The proposed round number is selected based on following condition:
+    - If the validator has received `ROUND CHANGE` messages from its peers, it picks the largest round number which has `F + 1` of `ROUND CHANGE` messages.
+    - Otherwise, it picks `1 + current round number` as the proposed round number.
+- Whenever a validator receives `F + 1` of `ROUND CHANGE` messages on the same proposed round number, it compares the received one with its own. If the received is larger, the validator broadcasts `ROUND CHANGE` message again with the received number.
+- Upon receiving `ceil(2N/3)` of `ROUND CHANGE` messages on the same proposed round number, the **validator** exits the round change loop, calculates the new **proposer**, and then enters `NEW ROUND` state.
+- Another condition that a validator jumps out of round change loop is when it receives verified block(s) through peer synchronization.
+
+#### Proposer selection
+
+Currently we support two policies: **round robin** and **sticky proposer**.
+
+- Round robin: in a round robin setting, proposer will change in every block and round change.
+- Sticky proposer: in a sticky proposer setting, proposer will change only when a round change happens.
+
+#### Validator list voting
+
+Istanbul BFT uses a similar validator voting mechanism as Clique and copies most of the content from Clique [EIP](https://github.com/ethereum/EIPs/issues/225). Every epoch transaction resets the validator voting, meaning if an authorization or de-authorization vote is still in progress, that voting process will be terminated.
```

Comment:
I think "every epoch transaction .... " is not very clear. something like 
"every epoch any pending votes are reset". 
is more accurate.

Reply:
Updated the documentation to reflect this